svm/tests/integration_test.rs

```diff
@@ -336,3 +336,200 @@ fn svm_integration() {
+
+    // fee payer
+    let mut account_data = AccountSharedData::default();
+    account_data.set_lamports(80_000);
+    mock_bank
+        .account_shared_data
+        .write()
+        .unwrap()
+        .insert(fee_payer, account_data.clone());
+    expected_inspected_accounts
+        .entry(fee_payer.clone())
+        .or_default()
+        .push((Some(account_data.clone()), true));
+
+    // sender
+    let mut account_data = AccountSharedData::default();
+    account_data.set_lamports(10_000_000);
+    mock_bank
+        .account_shared_data
+        .write()
+        .unwrap()
+        .insert(sender, account_data.clone());
+    expected_inspected_accounts
+        .entry(sender.clone())
+        .or_default()
+        .push((Some(account_data.clone()), true));
+
+    // recipient -- now alive
+    let mut account_data = AccountSharedData::default();
+    account_data.set_lamports(1_000_000);
+    mock_bank
+        .account_shared_data
+        .write()
+        .unwrap()
+        .insert(recipient, account_data.clone());
+    expected_inspected_accounts
+        .entry(recipient.clone())
+        .or_default()
+        .push((Some(account_data.clone()), true));
+
+    let mut transaction_builder = SanitizedTransactionBuilder::default();
+    transaction_builder.create_instruction(
+        transfer_program,
+        vec![
+            AccountMeta {
+                pubkey: sender,
+                is_signer: true,
+                is_writable: true,
+            },
+            AccountMeta {
+                pubkey: recipient,
+                is_signer: false,
+                is_writable: true,
+            },
+            AccountMeta {
+                pubkey: system,
+                is_signer: false,
+                is_writable: false,
+            },
+        ],
+        HashMap::from([(sender, Signature::new_unique())]),
+        456_u64.to_be_bytes().to_vec(),
+    );
+    let sanitized_transaction = transaction_builder
+        .build(Hash::default(), (fee_payer, Signature::new_unique()), true)
+        .unwrap();
+    let transaction_check = Ok(CheckedTransactionDetails {
+        nonce: None,
+        lamports_per_signature: 20,
+    });
+
+    // Load and execute the second transaction
+    let _result = batch_processor.load_and_execute_sanitized_transactions(
+        &mock_bank,
+        &[sanitized_transaction],
+        vec![transaction_check],
+        &TransactionProcessingEnvironment::default(),
+        &TransactionProcessingConfig::default(),
+    );
+
+    // Ensure all the expected inspected accounts were inspected
+    let actual_inspected_accounts = mock_bank.inspected_accounts.read().unwrap().clone();
+    for (expected_pubkey, expected_account) in expected_inspected_accounts {
+        let actual_account = actual_inspected_accounts.get(&expected_pubkey).unwrap();
+        assert_eq!(
+            expected_account, *actual_account,
+            "pubkey: {expected_pubkey}",
+        );
+    }
```

Comment:
Should also assert that `len` of actual and expected accounts is the same. In your test, the actual inspected accounts includes extra read only accounts.